### PR TITLE
Feature list of newsletters

### PR DESCRIPTION
**Pull request by The-Vector-007**

### CODE REVIEW

File: projects/commudle-admin/src/app/components/sidebar-menu/sidebar-menu.component.html [hunk 1]
```diff
@@ -96,6 +96,9 @@
         <nb-accordion-item-body *ngIf="isBadgesAdmin">
           <a [routerLink]="['/sys-admin', 'badges']">Badges</a>
         </nb-accordion-item-body>
+        <nb-accordion-item-body>
+          <a [routerLink]="['/newsletter-list']">Newsletters</a>
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/components/sidebar-menu/sidebar-menu.component.html]: Let's move this to the default menu on top

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.scss [hunk 2]
```diff
@@ -0,0 +1,26 @@
+@import './projects/commudle-admin/src/assets/styles/variables';
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.scss]: Please format this file

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  constructor(
+    private mainNewslettersService: MainNewslettersService
+  ) { }
+
+  ngOnInit(): void {
+    this.getPublishedNewsletters();
+  }
+
+  //ALERT: the string must always be in HTML format
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts]: Can there be other types of strings as input for this function?

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts]: Please format this file

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  //ALERT: the string must always be in HTML format
+  getFirstImageUrl(data: string): string {
+    let element = document.createElement('div');
+    element.innerHTML = data;
+    let images = element.getElementsByTagName('img');
+    return images[0].src;
+  }
+
+  getPublishedNewsletters() {
+    this.mainNewslettersService.publicIndex().subscribe((data) => {
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts]: Let's unsubscribe this

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.html [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            <div *ngFor="let newsletter of newsletters" class="newsletter-card com-md:w-1/3">
+              <div
+                class="
+                  com-flex com-flex-col com-justify-between com-h-full com-border-2 !com-border-gray-200 com-border-opacity-60 com-rounded-lg com-overflow-hidden
+                "
+              >
+                <img
+                  class="com-lg:h-48 com-md:h-36 com-w-full com-object-cover com-object-center"
+                  [src]="getFirstImageUrl(newsletter.content)"
+                  alt="blog"
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.html]: Change this alt to maybe title?

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.html [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                    [routerLink]="['/newsletters', newsletter.id]"
+                    class="
+                      !com-text-indigo-500
+                      com-inline-flex com-items-center
+                      com-md:mb-2
+                      com-lg:mb-0
+                      !com-no-underline
+                    "
+                    >Learn More
+                    <svg
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.html]: Let's replace this svg with icon

File: projects/commudle-admin/src/app/feature-modules/sys-admin/sys-admin-routing.module.ts [hunk 8]
```diff
@@ -79,4 +79,4 @@ const routes = [
   imports: [RouterModule.forChild(routes)],
   exports: [RouterModule],
 })
-export class SysAdminRoutingModule {}
+export class SysAdminRoutingModule {}
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/sys-admin/sys-admin-routing.module.ts]: Let's undo this change

File: projects/commudle-admin/src/app/app-routing.module.ts [hunk 9]
```diff
@@ -134,6 +135,10 @@ const routes = [
     path: 'newsletters/:main_newsletter_id',
     component: MainNewsletterComponent,
   },
+  {
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/app-routing.module.ts:137]: Let's convert the newsletter into a routing module and lazy load it here

**Review by Ashwin-op — CHANGES_REQUESTED** [review]

File: projects/commudle-admin/src/app/components/sidebar-menu/sidebar-menu.component.html (hunk 1 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/components/sidebar-menu/sidebar-menu.component.html]: moved

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.scss (hunk 2 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.scss]: done

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts (hunk 3 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts]: the input will always be in html format

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts (hunk 4 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts]: formatted

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts (hunk 5 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.ts]: unsubscribed

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.html (hunk 6 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.html]: changed

File: projects/commudle-admin/src/app/feature-modules/sys-admin/sys-admin-routing.module.ts (hunk 8 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/sys-admin/sys-admin-routing.module.ts]: done

File: projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.html (hunk 7 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/main-newsletters/components/main-newsletter-list/main-newsletter-list.component.html]: done

File: projects/commudle-admin/src/app/app-routing.module.ts (hunk 9 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/app-routing.module.ts:137]: done
